geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/IndexUsageInJoinQueryDistributedTest.java

```diff
@@ -0,0 +1,210 @@
+
+    // Create client regions
+    client.invoke(() -> createClientRegions());
+
+    // Load regions
+    int numProducts = 1000;
+    int numInstruments = 100;
+    client.invoke(() -> loadRegions(numProducts, numInstruments));
+
+    // Get number of deserializations before query
+    long deserializationsBeforeQuery = server.invoke(() -> getDeserializations());
+
+    // Execute query
+    client.invoke(() -> executeQuery(numInstruments));
+
+    // Get number of deserializations after query
+    long deserializationsAfterQuery = server.invoke(() -> getDeserializations());
+
+    // Verify number of deserializations during query
+    // 1 for each instrument in the instrumentTypesIndex (100)
+    // For each instrument, 1 product in the productIdKeyIndex (100)
+    // 1 for the bind parameter
+    assertThat(deserializationsAfterQuery - deserializationsBeforeQuery)
+        .isEqualTo((numInstruments * 2) + 1);
+  }
+
+  private void createServerRegionsAndIndexes()
+      throws RegionNotFoundException, IndexNameConflictException, IndexExistsException {
+    Cache cache = ClusterStartupRule.getCache();
+    QueryService queryService = cache.getQueryService();
+    Region productRegion = cache.createRegionFactory(REPLICATE).create(PRODUCT_REGION_NAME);
+    queryService.createKeyIndex("productIdKeyIndex", "productId", productRegion.getFullPath());
+    Region instrumentRegion = cache.createRegionFactory(REPLICATE).create(INSTRUMENT_REGION_NAME);
+    queryService.createIndex("instrumentTypesIndex", "types['TYPE']",
+        instrumentRegion.getFullPath());
+  }
+
+  private void createClientRegions() {
+    ClientCache cache = ClusterStartupRule.getClientCache();
+    cache.createClientRegionFactory(ClientRegionShortcut.PROXY).create(PRODUCT_REGION_NAME);
+    cache.createClientRegionFactory(ClientRegionShortcut.PROXY).create(INSTRUMENT_REGION_NAME);
+  }
+
+  private void loadRegions(int numProducts, int numInstruments) {
+    ClientCache cache = ClusterStartupRule.getClientCache();
+    Region productRegion = cache.getRegion(PRODUCT_REGION_NAME);
+    Region instrumentRegion = cache.getRegion(INSTRUMENT_REGION_NAME);
+
+    // Load product region
+    for (int i = 0; i < numProducts; i++) {
+      String productId = String.valueOf(i);
+      Product product = new Product(productId);
+      productRegion.put(productId, product);
+    }
+
+    // Load instrument region
+    for (int i = 0; i < numInstruments; i++) {
+      String instrumentId = String.valueOf(i);
+      Map<String, String> types = new HashMap<>();
+      types.put("TYPE", "TYPE1");
+      Instrument instrument = new Instrument(instrumentId, "0", types);
+      instrumentRegion.put(instrumentId, instrument);
+    }
+  }
+
+  private void executeQuery(int numInstruments) throws Exception {
+    String queryString =
+        "<trace> select i,p from /instrument i, /product p where i.productId = p.productId and i.types['TYPE']=$1";
```

Comment:
Could the uses of "/" here be replaced with Region.SEPARATOR, to remain consistent with the rest of the codebase?

Reply:
Yes, thanks. I'll clean that up.

Reply:
Done.